src/ObjCRuntime/Runtime.CoreCLR.cs

```diff
@@ -286,7 +286,12 @@ static void StructureToPtr (object obj, IntPtr ptr)
 			if (obj == null)
 				return;
 
-			Marshal.StructureToPtr (obj, ptr, false);
+			if (obj is bool b) {
+				// Only write a single byte for bools
+				Marshal.WriteByte (ptr, b ? (byte) 1 : (byte) 0);
```

Comment:
Can't this be specialized for bool, as in, provide an overload that takes bool?

Reply:
You mean add a `StructureToPtr (bool value, IntPtr ptr)` overload (in this class)?

Reply:
just checked, it won't matter: https://github.com/rolfbjarne/xamarin-macios/blob/6438efd805beb8a9f8c92d23b93e1040d14ebf36/src/ObjCRuntime/Runtime.CoreCLR.cs#L252